test/unit/datasets/test_datasets.py

```diff
@@ -38,3 +40,26 @@ def test_dataset_as_X_y(self, as_frame, fetch_function):
         assert isinstance(X, pd.DataFrame if as_frame else np.ndarray)
         assert y is not None
         assert isinstance(y, pd.Series if as_frame else np.ndarray)
+
+    def test_synthetic_datasets(self):
+        """Ensure that dataset creation is deterministic."""
+        rng = np.random.RandomState(12345)
+        X, y, gender = make_synthetic_dataset(random_state=rng)
+
+        X_train, _, y_train, _, gender_train, _ = train_test_split(
+            X, y, gender, test_size=0.3, random_state=rng
+        )
+
+        assert np.sum(X_train[0] < 0) == 8
```

Comment:
Are these two lines just random aggregate stats you found? If so, perhaps asserting the dimensions and mean of the features and targets is useful?